src/views/account/Grades/Modals/GradeReaction.tsx

```diff
@@ -0,0 +1,323 @@
+import { captureRef } from "react-native-view-shot";
+import { Screen } from "@/router/helpers/types";
+import { getSubjectData } from "@/services/shared/Subject";
+import { useGradesStore } from "@/stores/grades";
+import { Reel } from "@/services/shared/Reel";
+import PapillonSpinner from "@/components/Global/PapillonSpinner";
+
+// Types
+interface SubjectData {
+  color: string;
+  pretty: string;
+  emoji: string;
+}
+
+interface Grade {
+  id: string;
+  student: { value: number | null };
+  outOf: { value: number | null };
+  coefficient: number | null;
+  subjectName: string;
+  timestamp: number;
+}
+
+// Helper Functions
+const convertToBase64 = async (uri: string): Promise<string> => {
+  const response = await fetch(uri);
+  const blob = await response.blob();
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = () => resolve((reader.result as string).split(",")[1]);
+    reader.onerror = reject;
+    reader.readAsDataURL(blob);
+  });
+};
+
+const createReel = async (
+  grade: Grade,
+  imageUri: string,
+  imageWithoutEffectUri: string
+): Promise<Reel> => {
+  const [image, imageWithoutEffect] = await Promise.all([
+    convertToBase64(imageUri),
+    convertToBase64(imageWithoutEffectUri)
+  ]);
+
+  return {
+    id: grade.id,
+    timestamp: Date.now(),
+    image,
+    imagewithouteffect: imageWithoutEffect,
+    subjectdata: getSubjectData(grade.subjectName),
+    grade: {
+      value: grade.student.value?.toString() ?? "",
+      outOf: grade.outOf.value?.toString() ?? "",
+      coef: grade.coefficient?.toString() ?? "",
+    }
+  };
+};
+
+const GradeReaction: Screen<"GradeReaction"> = ({ navigation, route }) => {
+  const inset = useSafeAreaInsets();
+  const [mediaLibraryPermission, requestMediaLibraryPermission] = MediaLibrary.usePermissions();
+  const [cameraPermission, requestCameraPermission] = useCameraPermissions();
+  const cameraRef = useRef<CameraView | null>(null);
+  const composerRef = useRef(null);
+  const [capturedImage, setCapturedImage] = useState<string>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [grade] = useState<Grade>(route.params.grade);
+  const [subjectData, setSubjectData] = useState<SubjectData>({
+    color: "#888888",
+    pretty: "Matière inconnue",
+    emoji: "❓",
+  });
+
+  // Setup permissions
+  useEffect(() => {
+    const setupPermissions = async () => {
+      if (mediaLibraryPermission?.status !== PermissionStatus.GRANTED) {
+        await requestMediaLibraryPermission();
+      }
+      if (cameraPermission?.status !== PermissionStatus.GRANTED) {
+        await requestCameraPermission();
+      }
+    };
+    setupPermissions();
+  }, [mediaLibraryPermission, cameraPermission, requestMediaLibraryPermission, requestCameraPermission]);
+
+  // Fetch subject data
+  useEffect(() => {
+    setSubjectData(getSubjectData(grade.subjectName));
+  }, [grade.subjectName]);
+
+  React.useLayoutEffect(() => {

```

Comment:
Autant l'importer direct depuis React

```suggestion
  useLayoutEffect(() => {
```